mono/mini/mini-exceptions.c

```diff
@@ -664,10 +664,14 @@ unwinder_unwind_frame (Unwinder *unwinder,
 					   StackFrameInfo *frame)
 {
 	if (unwinder->in_interp) {
-		unwinder->in_interp = mono_interp_frame_iter_next (&unwinder->interp_iter, frame);
-		if (!unwinder->in_interp) {
+		gpointer new_sp;
+		memcpy (new_ctx, ctx, sizeof (MonoContext));
+		unwinder->in_interp = mono_interp_frame_iter_next (&unwinder->interp_iter, frame, &new_sp);
+		if (frame->type == FRAME_TYPE_INTERP)
+			/* This is needed so code which uses ctx->sp for frame ordering would work */
+			MONO_CONTEXT_SET_SP (new_ctx, new_sp);
```

Comment:
`new_sp` is a weird name, if it's really just the parent frame.  I kinda understand where it comes from, but let's not propagate that naming further than needed.